src/test/versionHandlers.test.ts

```diff
@@ -91,7 +101,161 @@ const v2Data = {
     version: 2,
 };
 
+const typeMappingWithGeo = {
+    "0": {
+        name: "Actin",
+        geometry: {
+            displayType: "PDB" as PdbDisplayType,
+            color: "#fff",
+            url: "url-to-data",
+        },
+    },
+    "1": {
+        name: "Budding vesicle",
+        geometry: {
+            displayType: "PDB" as PdbDisplayType,
+            color: "#fff",
+            url: "url-to-data",
+        },
+    },
+};
+
+const typeMappingWithDefaultGeo = {
+    "0": {
+        name: "Actin",
+        geometry: {
+            displayType: "SPHERE" as SphereDisplayType,
+            color: "",
+            url: "",
+        },
+    },
+    "1": {
+        name: "Budding vesicle",
+        geometry: {
+            displayType: "SPHERE" as SphereDisplayType,
+            color: "",
+            url: "",
+        },
+    },
+};
+
+const typeMappingNoGeo = {
+    "0": {
+        name: "Actin",
+    },
+    "1": {
+        name: "Budding vesicle",
+    },
+};
+
+const typeMappingMissingDisplayType = {
+    "0": {
+        name: "Actin",
+        geometry: {
+            color: "",
+            url: "url",
+        },
+    },
+    "1": {
+        name: "Budding vesicle",
+        geometry: {
+            color: "",
+            url: "url",
+        },
+    },
+};
+
+const typeMappingMissingUrl = {
+    "0": {
+        name: "Actin",
+        geometry: {
+            color: "",
+            displayType: "OBJ" as ObjDisplayType,
+            url: "",
+        },
+    },
+    "1": {
+        name: "Budding vesicle",
+        geometry: {
+            color: "",
+            displayType: "PDB" as PdbDisplayType,
+            url: "",
+        },
+    },
+};
+
+const v3Data = {
+    ...v2Data,
+    typeMapping: typeMappingWithDefaultGeo,
+    version: 3,
+};
+
 describe("Version handlers", () => {
+    describe("makeUrlErrorMessage", () => {
+        test("it will create a message for the user if there is not displayType", () => {
```

Comment:
This is a little confusing because the tested function doesn't actually know about the existence of `displayType`. I think it would be better to say something like "it will create a message for the user given an empty string for the url" and then "it will create a message for the user given a non-empty string for the url" for the following test